### PR TITLE
Avoid performing fusion on operations with `flow.dispatch.region` or `flow.dispatch.workgroups` operations.

### PR DESCRIPTION
**Pull request by MaheshRavishankar**
Towards #14337

### CODE REVIEW

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp [hunk 1]
```diff
@@ -197,6 +198,9 @@ static FailureOr<unsigned> fuseMultiUseProducers(Operation *funcOp,
   unsigned numCandidates = 0;
   OpBuilder builder(context);
   funcOp->walk<WalkOrder::PreOrder>([&](linalg::GenericOp genericOp) {
+    if (!isNonNullAndOutsideDispatch(genericOp)) {
```
Comment (pzread) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp:201]: (Optional) Should we use the same `fuseElementwiseOpsControlFn` to also control the multi-use fusion?

**Review by pzread — APPROVED** [review]
I might not follow the latest development #14337, so just curious why we will have dispatch region ops at elementwise fusion stage? I thought we only formed dispatches after that.

**Comment by MaheshRavishankar** [comment]
> I might not follow the latest development #14337, so just curious why we will have dispatch region ops at elementwise fusion stage? I thought we only formed dispatches after that.

This is meant to support cases where dispatches has been pre-formed to override the default heuristics of fusion that IREE uses.

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp (hunk 1 above)
Comment (MaheshRavishankar) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp:201]: I havent looked into that, but at the high level I think the requirements are a bit different for non-multi-use fusion and multi-use fusion....

**Pull request merged**